iot-hub/Samples/device/DeviceReconnectionSample/DeviceReconnectionSample.cs

```diff
@@ -25,10 +25,10 @@ public class DeviceReconnectionSample
         private readonly TransportType _transportType;
 
         private readonly ILogger _logger;
-        private static DeviceClient s_deviceClient;
 
-        private static ConnectionStatus s_connectionStatus;
-        private static bool s_wasEverConnected;
+        // Mark these fields as volatile so that their latest values are referenced.
+        private static volatile DeviceClient s_deviceClient;
```

Comment:
why volatile? 
Is the value changing in sample below?

Reply:
yes, the device client will get reinitialized whenever it goes into "Disconnected" state -> depending on the exception